testing/test_metannotate_end_to_end.sh

```diff
@@ -14,38 +14,64 @@ python run_metannotate.py --orf_files=data/MetagenomeTest.fa --hmm_files=data/hm
 
 echo "Verifying outputs..."
 
-cd testing/test_constants
+EXIT_CODE=0
+ROOT_PATH=$(pwd)
 
-# Get reference hash file checksums
-FIRST_HASH=$(md5sum rpoB_0_msa_0.fa | cut -d ' ' -f 1)
-SECOND_HASH=$(md5sum rpoB_0_refseq_msa_1.fa | cut -d ' ' -f 1)
+## Check that the number of files in --output_dir is correct
+# TODO: replace this test with individual tests for each expected file
+NUM_OUTPUT_FILES=$(ls ${ROOT_PATH}/test_output | wc -l | sed 's/ //g')
+CORRECT_NUM_FILES=13
+if [[ ${NUM_OUTPUT_FILES} -eq ${CORRECT_NUM_FILES} ]]; then
+    echo "Correct number of files in output_dir ("${CORRECT_NUM_FILES}")"
+else
+    echo "Unexpected number of files in output_dir. Expected: "${CORRECT_NUM_FILES}"; Produced: "${NUM_OUTPUT_FILES}
+    EXIT_CODE=1
+fi
 
-cd ../..
+## Check hashes
+# Get reference hash file checksums
+declare -a REFERENCE_FILE_HASHES
+cd ${ROOT_PATH}/testing/test_constants
+REFERENCE_FILE_HASHES[0]=$(md5sum rpoB_0_msa_0.fa | cut -d ' ' -f 1)
+REFERENCE_FILE_HASHES[1]=$(md5sum rpoB_0_refseq_msa_1.fa | cut -d ' ' -f 1)
+# only first 14 columns of annotations table are consistent between runs
+REFERENCE_FILE_HASHES[2]=$(cut -f 1-14 rpoB_0_MetagenomeTest_0_annotations.tsv | md5sum | cut -d ' ' -f 1)
 
 # Store generated test output hashes in array
+# Keys in REFERENCE_FILE_HASHES should correspond to same key in REGENERATED_FILE_HASHES
```

Comment:
By ```keys``` do you mean ```elements```?

Reply:
Also its good practice to not use variable names (unless you have to) in comments. The comments should use plain languages. "The reference file hashes should match the input file hashes."

Reply:
The comments should explain what the code is trying to achieve. Not what it does.

Reply:
Might want to remove this comment (just saw the below). The comment should be something like "Store the output file hashes for later comparison"